goplay/src/main/java/be/tapped/goplay/content/SearchRepo.kt

```diff
@@ -2,61 +2,20 @@ package be.tapped.goplay.content
 
 import arrow.core.Either
 import arrow.core.computations.either
-import be.tapped.common.internal.executeAsync
 import be.tapped.goplay.ApiResponse
-import be.tapped.goplay.common.safeBodyString
-import be.tapped.goplay.common.vierVijfZesApi
-import kotlinx.serialization.decodeFromString
-import kotlinx.serialization.json.Json
-import kotlinx.serialization.json.JsonObject
-import kotlinx.serialization.json.add
-import kotlinx.serialization.json.buildJsonArray
-import kotlinx.serialization.json.buildJsonObject
-import kotlinx.serialization.json.decodeFromJsonElement
-import kotlinx.serialization.json.jsonArray
-import kotlinx.serialization.json.jsonObject
-import kotlinx.serialization.json.put
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.RequestBody.Companion.toRequestBody
-
-internal class JsonSearchResultsParser {
-    fun parse(json: String): Either<ApiResponse.Failure, List<SearchHit>> = Either.catch {
-        val hitsArr = Json.decodeFromString<JsonObject>(json)["hits"]!!.jsonObject["hits"]!!.jsonArray
-        Json.decodeFromJsonElement<List<SearchHit>>(hitsArr)
-    }.mapLeft(ApiResponse.Failure::JsonParsingException)
-}
+import be.tapped.goplay.common.ktorClient
+import io.ktor.client.HttpClient
 
 public sealed interface SearchRepo {
     public suspend fun search(query: String): Either<ApiResponse.Failure, ApiResponse.Success.Content.SearchResults>
 }
 
-internal class HttpSearchRepo(
-    private val client: OkHttpClient,
-    private val jsonSearchResultsParser: JsonSearchResultsParser,
-) : SearchRepo {
+internal class HttpSearchRepo(private val client: HttpClient = ktorClient) : SearchRepo {
```

Comment:
Nit: not huge fan of the default arguments personally but we probably need to figure out some DI system, with `expect/actual` as well to make all this work with Kotlin MPP in a nice way.

Reply:
I fully agree.

For now I added the default just to make it more easily testable.
We can definitely improve this!